frontend/src/metabase/dashboard/containers/CreateDashboardModal.unit.spec.tsx

```diff
@@ -38,7 +49,15 @@ function setup({ mockCreateDashboardResponse = true } = {}) {
       .reply(200, (url, body) => body);
   }
 
-  renderWithProviders(<CreateDashboardModal onClose={onClose} />);
+  renderWithProviders(<CreateDashboardModal onClose={onClose} />, {
+    storeInitialState: {
```

Comment:
why don't we need to have `storeInitialState` earlier?

Reply:
I mean in the previous version, we don't need to pass `storeInitialState`. So I'm not sure why do we need to pass it now.

Does that question make sense?

Reply:
Oh, my bad, thanks for explaining!

So what's going on here is that we test `CreateDashboardModal` and it has a collection picker. The collection picker tries to pick a smart default collection using the [`getInitialCollectionId` selector](https://github.com/metabase/metabase/blob/304552fd229d64f4845df9406b4d5a76edcf0158/frontend/src/metabase/entities/collections/getInitialCollectionId.ts#L42). It checks various stuff, but in this specific case we're always running into a fallback behavior which is "Our analytics" for admin users or user's personal collection for non-admins. User info is grabbed from Redux as well, so it affects our tests. `renderWithProviders` was using this custom [`getUser` helper](https://github.com/metabase/metabase/blob/6d7f778ef68761b9af943d07ed868d9d0f67f64a/frontend/test/__support__/ui.js#L18) and we've switched to [`createMockUser`](https://github.com/metabase/metabase/blob/6d7f778ef68761b9af943d07ed868d9d0f67f64a/frontend/src/metabase-types/api/mocks/user.ts#L3). The issue is that `getUser` returned a user without `personal_collection_id` and `createMockUser` returns a user with this property set to `1`. Here's what happened next:

1. `CreateDashboardModal` gets rendered and `getInitialCollectionId` gets called
2. The selector can't come up with any smart collections suggestions, so it goes to a fallback behavior
3. The `currentUser` object in Redux is an admin, so it makes sense to offer Our analytics to them
4. Oops, Our analytics isn't present in the entity store, so our [`can_write` check](https://github.com/metabase/metabase/blob/6d7f778ef68761b9af943d07ed868d9d0f67f64a/frontend/src/metabase/entities/collections/getInitialCollectionId.ts#L76) returns `false`, so we return the personal collection ID
5. Before we didn't have the personal collection ID, so `canonicalCollectionId` just returned `null` which is root collection's ID anyway

So here I'm putting our analytics into redux explicitly to avoid that. It gets loaded right after you open Metabase, so there shouldn't really be any cases when we don't have the root collections loaded

Reply:
Thank you for the thorough explanation! I really appreciate that there's so much thought behind such a simple change.